src/taco-utils/promisesUtils.ts

```diff
@@ -26,12 +30,101 @@ module TacoUtility {
         which are executed only for side effects, and are independent of the result of this method
         */
         public static wrapExecution<T>(initializer: () => void, codeToExecute: () => T,
-            finalizer: () => void, failHandler: (reason: any) => void = () => { /* Do Nothing */ } ): T {
+            finalizer: () => void, failHandler: (reason: any) => void = () => { /* Do Nothing */ }): T {
             initializer();
             var result = codeToExecute();
             Q(result).finally(finalizer).fail(failHandler);
             return result;
         }
+
+        /**
+         * Sequentially runs a number of promises obtained from an array of values
+         *  @value: array of values used to create chain of promises
+         *  @func: accumulator function which runs over each array value and returns a promise which resolves to an accumulated value
+         *  @initialValue: initial accumulated value
+         */
+        public static chain<T, U>(values: T[], func: (value: T, valueSoFar: U) => Q.Promise<U>, initialValue?: U): Q.Promise<U> {
+            return values.reduce(function(soFar: Q.Promise<U>, val: T): Q.Promise<U> {
+                return soFar.then(function(valueSoFar: U): Q.Promise<U> {
+                    return func(val, valueSoFar);
+                });
+            }, Q(initialValue || <U>{}));
```

Comment:
Why not just `Q(initialValue)`? If U is supposed to be a numeric type for example, then {} is incorrect. If you just leave it as initialValue, then it will go to undefined (which inhabits all types and the code should be willing to deal with).